Core/TransactionHandler.php

```diff
@@ -0,0 +1,314 @@
+     * @param SuccessResponse $oResponse
+     *
+     * @return null
+     *
+     * @since 1.0.1
+     */
+    private function _handleNotificationSuccess($oBackendService, $oResponse)
+    {
+        $aData = $oResponse->getData();
+        $oUtilsDate = Registry::getUtilsDate();
+        $sConvertedTimestamp = $oUtilsDate->formatDBTimestamp($oUtilsDate->formTime($aData['completion-time-stamp']));
+
+        // find root transaction ID
+        $sRootTransactionId = $this->_getRootTransactionId($oResponse->getParentTransactionId());
+
+        $oOrder = oxNew(Order::class);
+        if (!$oOrder->loadWithTransactionId($sRootTransactionId)) {
+            $this->_oLogger->error('No order found for transactionId: ' . $oResponse->getParentTransactionId());
+            return;
+        }
+
+        // transaction amount
+        $fAmount = $oResponse->getRequestedAmount()->getValue();
+
+        // determine if transaction state is closed
+        $sTransactionState = $oBackendService->isFinal($oResponse->getTransactionType()) ?
+                Transaction::STATE_CLOSED : $aData['transaction-state'];
+
+        // create an array with the properties to be saved in the transaction database entry
+        $aTransactionProps = array(
+            'ordernumber' => $oOrder->oxorder__oxordernr->value,
+            'orderid' => $oOrder->oxorder__oxid->value,
+            'transactionid' => $oResponse->getTransactionId(),
+            'parenttransactionid' => $oResponse->getParentTransactionId(),
+            'requestid' => $oResponse->getRequestId(),
+            'action' => $oOrder->getOrderPayment()->oxpayments__wdoxidee_transactionaction->value,
+            'type' => $oResponse->getTransactionType(),
+            'state' => $sTransactionState,
+            'amount' => $fAmount,
+            'currency' => $oResponse->getRequestedAmount()->getCurrency(),
+            'responsexml' => base64_encode($oResponse->getRawData()),
+            'date' => $sConvertedTimestamp
+        );
+
+        // add a transaction database entry from properties array
+        Transaction::createDbEntryFromArray($aTransactionProps);
+
+        // check if the parent transaction needs to be set to 'closed'
+        $this->updateParentTransactionStateIfNecessary($oResponse);
+    }
+
+    /**
+     * Checks if the parent transaction still has money to be captured.
+     * If no money is available anymore, the state is set to closed.
+     *
+     * @param SuccessResponse $oResponse
+     *
+     * @since 1.0.1
+     */
+    public function updateParentTransactionStateIfNecessary($oResponse)
+    {
+        $sParentTransactionId = $oResponse->getParentTransactionId();
+
+        $oParentTansaction = oxNew(Transaction::class);
+        $oParentTansaction->loadWithTransactionId($sParentTransactionId);
+
+        $fRestAmount = $this->getTransactionMaxAmount($sParentTransactionId);
+
+        if ($fRestAmount <= 0) {
+            // set transaction state to closed
```

Comment:
We can omit this comment as it is obvious enough.

Reply:
👍